onnxruntime/contrib_ops/cuda/cuda_contrib_kernels.cc

```diff
@@ -80,6 +80,13 @@ class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kMSDomain, 1
 class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kMSDomain, 1, float_int8_t, QAttention);
 class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kMSDomain, 1, MLFloat16_int8_t, QAttention);
 
+#if defined(CUDA_VERSION) && CUDA_VERSION >= 11000
```

Comment:
These are compile time constraints. What would it take to incorporate device properties into the code and make it environment aware?